concurrent_map.go

```diff
@@ -5,75 +5,114 @@ import (
 	"sync"
 )
 
-var SHARD_COUNT = 32
-
-// A "thread" safe map of type string:Anything.
-// To avoid lock bottlenecks this map is dived to several (SHARD_COUNT) map shards.
-type ConcurrentMap []*ConcurrentMapShared
-
-// A "thread" safe string to anything map.
-type ConcurrentMapShared struct {
-	items        map[string]interface{}
+// ShardCount represents the number of shards the map is divided into.
+var ShardCount = 32
+
+// ConcurrentMap is a "thread" safe map of type string:Anything.
+// To avoid lock bottlenecks this map is dived to several (ShardCount) map shards.
+type ConcurrentMap []*ConcurrentMapShard
+
+// ConcurrentMapShared is a "thread" safe string to anything map.
+type ConcurrentMapShard struct {
+	maxSize uint32
```

Comment:
maybe we should drop all uint32 usages. Should have ints instead and the code will get cleaner (no more cast to and from uint32)

Reply:
done